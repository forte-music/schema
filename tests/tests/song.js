import client from '../client';

it('should get the artists of a song', async () => {
  const variables = { songId: 'song:2:1' };
  const { song } = await client.request(
    `
      query($songId: ID!) {
        song(id: $songId) {
          id
          name

          artists {
            id
            name
          }
        }
      }
    `,
    variables
  );

  expect(song.artists.length).toBeGreaterThan(0);
});

it('should get all items when limit is -1', async () => {
  const { songs } = await client.request(
    `
      query {
        songs(input: { limit: -1 }) {
          count
          edges {
            cursor
            node {
              id
            }
          }
        }
      }
    `
  );

  expect(songs.count).toEqual(songs.edges.length);
});
