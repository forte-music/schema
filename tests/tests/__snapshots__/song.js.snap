// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get a song by id 1`] = `
Object {
  "song": Object {
    "album": Object {
      "artworkUrl": "http://is4.mzstatic.com/image/thumb/Music5/v4/08/da/96/08da9619-3f9b-7c95-60d1-6c18cfdd4dbd/source/600x600bb.jpg",
      "duration": 211,
      "id": "album:2",
      "name": "I'm That (Remix)",
      "releaseYear": 2015,
    },
    "artists": Array [
      Object {
        "id": "artist:2",
        "name": "R. City",
      },
      Object {
        "id": "artist:21",
        "name": "Beenie Man",
      },
      Object {
        "id": "artist:22",
        "name": "Azealia Banks",
      },
    ],
    "diskNumber": 1,
    "id": "song:2:1",
    "name": "I'm That (Remix)",
    "stats": Object {
      "id": "song:2:1:stats:song",
      "liked": false,
      "stats": Object {
        "id": "song:2:1:stats",
        "lastPlayed": null,
        "playCount": 0,
      },
    },
    "streamUrl": "/music/01 - I'm That... (Remix) [feat. Beenie Man & Azealia Banks].m4a",
    "trackNumber": 1,
  },
}
`;
